src/services/services.ts

```diff
@@ -5008,6 +5015,97 @@ namespace ts {
             return getDefinitionFromSymbol(type.symbol, node);
         }
 
+         /// Goto implementation
+        function getImplementationAtPosition(fileName: string, position: number): ImplementationLocation[] {
+            synchronizeHostData();
+
+            const node = getTouchingPropertyName(getValidSourceFile(fileName), position);
+            const typeChecker = program.getTypeChecker();
+
+            // If invoked directly on a shorthand property assignment, then return
+            // the declaration of the symbol being assigned (not the symbol being assigned to).
+            if (node.parent.kind === SyntaxKind.ShorthandPropertyAssignment) {
+                return getReferenceEntryForShorthandPropertyAssignment(node, typeChecker);
+            }
+            else if (node.kind === SyntaxKind.SuperKeyword || isSuperProperty(node.parent)) {

```

Comment:
`this` does not need to be a special case; it works fine with the find all references implementation. Previously I was being too narrow with what was returned. Properties on `super` always point to one implementation, but properties on `this` should really point to all implementations in subclasses as well as the current class.